tests/glsl-intrinsic/atomic/atomicStorageBuffer.slang

```diff
@@ -0,0 +1,318 @@
+
+        && (float16Buffer.data[0] = half(5)) == half(5)
+        && atomicMax(float16Buffer.data[0], half(1)) == half(5)
+        && float16Buffer.data[0] == half(5)
+
+        && (float16Buffer.data[0] = half(5)) == half(5)
+        && atomicExchange(float16Buffer.data[0], half(2)) == half(5)
+        && float16Buffer.data[0] == half(2)
+#endif // TEST_when_shader_atomic_float2_is_available
+        ;
+}
+
+bool testAtomicFloat32()
+{
+    return true
+
+        && (float32Buffer.data[0] = float(5)) == float(5)
+        && atomicAdd(float32Buffer.data[0], float(1)) == float(5)
+        && float32Buffer.data[0] == float(6)
+
+#ifdef TEST_when_shader_atomic_float2_is_available
+        && (float32Buffer.data[0] = float(5)) == float(5)
+        && atomicMin(float32Buffer.data[0], float(1)) == float(5)
+        && float32Buffer.data[0] == float(1)
+
+        && (float32Buffer.data[0] = float(5)) == float(5)
+        && atomicMax(float32Buffer.data[0], float(1)) == float(5)
+        && float32Buffer.data[0] == float(5)
+
+        && (float32Buffer.data[0] = float(5)) == float(5)
+        && atomicExchange(float32Buffer.data[0], float(2)) == float(5)
+        && float32Buffer.data[0] == float(2)
+#endif // TEST_when_shader_atomic_float2_is_available
+        ;
+}
+
+bool testAtomicFloat64()
+{
+    return true
+
+        && (float64Buffer.data[0] = double(5)) == double(5)
+        && atomicAdd(float64Buffer.data[0], double(1)) == double(5)
+        && float64Buffer.data[0] == double(6)
+
+#ifdef TEST_when_shader_atomic_float2_is_available
+        && (float64Buffer.data[0] = double(5)) == double(5)
+        && atomicMin(float64Buffer.data[0], double(1)) == double(5)
+        && float64Buffer.data[0] == double(1)
+
+        && (float64Buffer.data[0] = double(5)) == double(5)
+        && atomicMax(float64Buffer.data[0], double(1)) == double(5)
+        && float64Buffer.data[0] == double(5)
+
+        && (float64Buffer.data[0] = double(5)) == double(5)
+        && atomicExchange(float64Buffer.data[0], double(2)) == double(5)
+        && float64Buffer.data[0] == double(2)
+#endif // TEST_when_shader_atomic_float2_is_available
+        ;
+}
+
+layout(local_size_x = 1) in;
+void computeMain()
+{
+    // testing has the following pattern in groups:
+    // set, op, test
+    // this is to remove dependency on other ops when testing
+    outputBuffer.data[0] = true
+            && testAtomicInt32()
+            && testAtomicInt64()
+            && testAtomicUint32()
+            && testAtomicUint64()
+            && testAtomicFloat16()
+            && testAtomicFloat32()
+            && testAtomicFloat64()
+            ;    
+    // to ensure these buffers don't mess-up result
+    int32Buffer.data[0] = 2;
+    int64Buffer.data[0] = 2;
+    uint32Buffer.data[0] = 2;
+    uint64Buffer.data[0] = 2;
+    float16Buffer.data[0] = half(2.0f);
+    float32Buffer.data[0] = float(2.0f);
+    float64Buffer.data[0] = double(2.0);
```

Comment:
I am not sure why these lines are needed.

Reply:
when checking the result buffer so there is only one value that is the value 1 (the results take all buffer outputs into account)